core/components/userimport/model/userimport/importhandler.class.php

```diff
@@ -557,6 +560,58 @@ private function _saveUser($row, $fieldvalues, $groups, $role, $autoUsername, $s
     }
     
     /**
+     * Looks for a field called "password" in the import file, validates it, and defaults to an auto-generated password
+     *
+     * @param modUser $user
+     * @param array $fieldvalues
+     * @param int $row
+     * @return bool|string
+     */
+    private function _setPassword(modUser &$user, $fieldvalues, $row) {
+
+        if(array_key_exists('password',$fieldvalues)) {
+            if($providedPassword = $this->_validateProvidedPassword($fieldvalues['password'],$row)) {
+                $password = $providedPassword;
+            }
+            else {
+                $this->modx->log(modX::LOG_LEVEL_WARN, '-> '.$this->modx->lexicon('userimport.import_users_row').$row.' '.$this->modx->lexicon('userimport.import_users_log_err_password_ignored'));
+                $password = $user->generatePassword(8);
+            }
+        }
+        else {
+            $password = $user->generatePassword(8);
+        }
+
+        return $password;
+
+    }
+
+    /**
+     * Check that the password provided in the import file is suitable for use
+     *
+     * @param $password
+     * @param $row
+     * @return bool|string Returns false if not valid
+     */
+    private function _validateProvidedPassword($password,$row) {
+        
+        if(!is_string($password)) {
```

Comment:
@prioritypie First of all - thanks for the PR! Just refactoring your code a little bit and asking why this shouldn't be a string here? I'm thinking of simply typecasting the password field. What was your intention here? Maybe I'm overlooking something.